namespace.c

```diff
@@ -1007,6 +1023,18 @@ rb_initialize_main_namespace(void)
     rb_const_set(rb_cNamespace, rb_intern("MAIN"), main_ns);
 
     vm->main_namespace = th->ns = main_namespace = ns;
+
+    // If RubyGems was loaded in the root namespace during boot,
+    // remove it and load a fresh copy in the main namespace
```

Comment:
The line `th->ns = ns` specifies the main namespace as the current namespace. So the cleaning RubyGems status should be done above the line.